src/arrayMethodJoin.js

```diff
@@ -4,8 +4,27 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (this.length === 0) {
+      return '';
+    }
+
+    let result = '';
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (typeof this[i] === 'undefined' || this[i] === null) {
+        result += separator;
+      } else {
+        result += this[i];
+        result += separator;
+      }
+    }
```

Comment:
great solution, but try to shorten your code because you use `result += separator;` twice.